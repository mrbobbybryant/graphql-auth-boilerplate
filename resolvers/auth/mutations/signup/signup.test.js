import { signup } from './index';

const create = jest.fn();
const jwt = jest.fn();
const User = {
  create,
};

const testSignup = signup(User, jwt);

describe('signup Tests', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  test('Should return an error if the password is not long enough.', async () => {
    try {
      const result = await testSignup(
        {},
        {
          input: {
            email: 'bobby@crossfield.com',
            password: '1234',
          },
        },
        {},
      );
      throw 'This should not run';
    } catch (error) {
      const info = JSON.parse(error.message);
      expect(info.message).toEqual('The password must be 8 characters.');
      expect(info.code).toEqual(400);
      expect(info.field).toEqual('password');
    }
  });

  test('Should return an error if the email is not valid.', async () => {
    try {
      const result = await testSignup(
        {},
        {
          input: {
            email: 'bobbycrossfield.com',
            password: '12345678',
          },
        },
        {},
      );
      throw 'This should not run';
    } catch (error) {
      const info = JSON.parse(error.message);
      expect(info.message).toEqual('The email format is invalid.');
      expect(info.code).toEqual(400);
      expect(info.field).toEqual('email');
    }
  });

  test('Should return an error if the email is already taken.', async () => {
    create.mockRejectedValueOnce({ constraint: 'users_email_unique' });
    try {
      const result = await testSignup(
        {},
        {
          input: {
            email: 'bobby@crossfield.com',
            password: '12345678',
          },
        },
        {},
      );
      throw 'This should not run';
    } catch (error) {
      const info = JSON.parse(error.message);
      expect(info.message).toEqual('A user with this email already exists.');
      expect(info.code).toEqual(400);
      expect(info.field).toEqual('email');
    }
  });

  test('Should work.', async () => {
    create.mockReturnValueOnce({ id: 1, email: 'bobby@crossfield.com' });
    jwt.mockReturnValueOnce('UIHUIHUI');
    const result = await testSignup(
      {},
      {
        input: {
          email: 'bobby@crossfield.com',
          password: '12345678',
        },
      },
      {},
    );

    expect(result).toMatchSnapshot();
  });
});
