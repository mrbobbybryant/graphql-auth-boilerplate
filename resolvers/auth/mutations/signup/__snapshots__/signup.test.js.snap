// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signup Tests Should work. 1`] = `
Object {
  "email": "bobby@crossfield.com",
  "id": 1,
  "token": "UIHUIHUI",
}
`;
